tools/c7n_tencentcloud/c7n_tencentcloud/resources/cvm.py

```diff
@@ -17,5 +25,81 @@ class resource_type(ResourceTypeInfo):
         version = "2017-03-12"
         enum_spec = ("DescribeInstances", "Response.InstanceSet[]", {})
         metrics_instance_id_name = "InstanceId"
-        paging_def = {"method": PageMethod.Offset, "limit": {"Key": "Limit", "value": 20}}
+        paging_def = {"method": PageMethod.Offset, "limit": {"key": "Limit", "value": 20}}
         resource_preifx = "instance"
+        taggable = True
+        batch_size = 10
+
+
+class CvmAction(TencentCloudBaseAction):
+    schema_alias = True
+
+    """cvm base api_method_name """
+
+    def process(self, resources):
+        for batch in chunks(resources, self.resource_type.batch_size):
+            params = self.get_request_params(batch)
+            if params is not None:
+                self.do_request(params)
+
+    def do_request(self, params):
+        """process_cvm"""
+        try:
+            client = self.get_client()
+            resp = client.execute_query(self.t_api_method_name, params)
+            failed_resources = jmespath.search("Response.Error", resp)
+            if failed_resources is not None:
+                raise PolicyExecutionError(f"{self.data.get('type')} error")
+            self.log.debug("%s resources: %s, cvm: %s",
+                           self.data.get('type'),
+                           params['InstanceIds'],
+                           params)
+        except (RetryError, TencentCloudSDKException) as err:
+            raise PolicyExecutionError(err) from err
+
+    def get_request_params(self, resources):
+        """
+        The default value returns InstanceIds , if there is customization,
+        it will be implemented in subclasses
+        """
+        ids = []
+        for resource in resources:
+            ids.append(resource[self.resource_type.id])
+        if len(ids) > 0:
+            return {"InstanceIds": ids}
+        return None
+
+
+@CVM.action_registry.register('stop')
+class CvmStopAction(CvmAction):
+    """stop_cvm"""
+    schema = type_schema("stop")
+    t_api_method_name = "StopInstances"
+
+    def get_request_params(self, resources):
+        """get_params_stop"""
+        ids = []
+        for resource in resources:
```

Comment:
this correct state filtering can be done with self.filter_resources (c7n/element) in process, and it will also log the implicit filtering in the action.